JavaScript/JavaScriptSDK/TelemetryContext.ts

```diff
@@ -128,13 +130,21 @@ module Microsoft.ApplicationInsights {
             this._applyDeviceContext(envelope, this.device);
             this._applyInternalContext(envelope, this.internal);
             this._applyLocationContext(envelope, this.location);
-            this._applyOperationContext(envelope, this.operation);
             this._applySampleContext(envelope, this.sample);
             this._applyUserContext(envelope, this.user);
+            this._applyOperationContext(envelope, this.operation);
 
             envelope.iKey = this._config.instrumentationKey();
 
-            this._sender.send(envelope);
+            if (this.sample.isSampledIn(envelope)) {
+                this._sender.send(envelope);
+            }
+            else {
+                _InternalLogging.logInternalMessage(LoggingSeverity.WARNING,
+                    "Telemetry is sampled and not sent to the AI service. SampleRate is " + this.sample.sampleRate);
```

Comment:
Do we really need this, unclear what customer benefit is. Will this appear in browser console, probably not what we want